.github/workflows/neuron-ci.yml

```diff
@@ -151,7 +151,7 @@ jobs:
               echo "CC="$CC
               echo "CXX="$CXX
               cmake $CMAKE_OPTION  -DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX -DNRN_ENABLE_TESTS=ON -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR ..;
-              cmake --build . -- -j;
+              cmake --build .
```

Comment:
it would be nice to build in parallel. cmake accepts `--parallel`

Reply:
The problem is that GitHub Actions jobs' compiler processes were being killed. Do you think `cmake --build . --parallel` will manage resources more intelligently than `cmake --build . -- -j`?

Reply:
Also, IIUC it was only added in CMake v3.12 and it looks like [we request an older version in some CI jobs](https://github.com/neuronsimulator/nrn/blob/99a6a370ef143b3c681eda1deea68bc7e2b8f21d/.github/workflows/neuron-ci.yml#L21), and take the default Ubuntu 18.04 version in others; I think that's v3.10.

We could try an explicit `cmake --build . -- -j 2` to see if the problem is related to the "smart" default parallelism, but I wonder if that should be a followup and we should merge this to get CIs green again.

Reply:
lets do this in subsequent PR after Rebase (to avoid longer build time)